### ISSUE TITLE
Add tips to polkadot

### ISSUE THREAD

**Issue by dandanlen** [chainflip-io/chainflip-backend#3031]
After #3017 it's a little tricky to integrate tips nicely in the extrinsic builder.

It might be worth re-thinking the ApiCall trait a bit. 

If the runtime version is updated, it invalidates the payload. 
If the tip is updated, it doesn't invalidate the payload.

---

This seems to be incompatible with polkadot tips (the tx would be flagged as invalid if we update the tip). I've disabled adding the tip for now, will add it back in in another PR.

_Originally posted by @dandanlen in https://github.com/chainflip-io/chainflip-backend/issues/3017#issuecomment-1491948235_
